healvis/observatory.py

```diff
@@ -219,9 +249,19 @@ def get_observed_region(self, Nside):
     def vis_calc(self, pcents, tinds, shell, vis_array, Nfin, beam_pol='pI'):
         if len(pcents) == 0:
             return
+
+        haspoles = True
+        if not hasattr(self, 'north_poles'):
```

Comment:
how would it be possible that the instance has no attribute `north_poles`? Is it possible to _not_ call `get_uvw`?

Reply:
In a recent change (initializing all instance variables in the __init__) the north_poles attribute is set to None, so I changed this to check if it isn't None before proceeding.